### ISSUE TITLE
The name of the class X should end with the suffix Component

### ISSUE THREAD

**Issue by muuvmuuv** [angular-eslint/angular-eslint#90]
I totally agree with that rule and it is really important but I prefer a slightly different style. 

In my app every component that is used within anything is named xx.component.ts and mostly shared (through a shared module) and isolated. Then I have components that are "pages", those are entry components for routes and are unique. This has a few benefits, the first is that I can see way faster that this is either a component or just a page that holds components, it is more readable and separates those from each other when I include them in a module.

TL;DR: I would like to add my own style; xx.page.ts must end with "Page" and xxx.component.ts must end with "Component".

**Comment by bbarry** [comment]
I would like to see a more general alternative rule like:

any file matching the pattern `/\.([a-z]+)\.ts$/` defines a class type suffix
should have 1 exported named symbol matching the file name in PascalCase after the characters `-` and `.` are removed and the file extension dropped
and no other classes in the file with this suffix

I don't think this rule really belongs in this particular repository. Perhaps it should apply globally to eslint or as a plugin rule on its own?

**Comment by JamesHenry** [comment]
Thank you for submitting the issue, please see https://github.com/angular-eslint/angular-eslint/issues/86#issuecomment-661665652 for a brief personal update. I will endeavour to review this in the near future, if one of my co-contributors does not get to it first.

**Comment by JamesHenry** [comment]
I think I agree with @bbarry's take that, for now at least, this is better suited to a custom plugin for a particular workspace, rather than in this core toolset.

Happy to reconsider if the rule gets created and published and in general custom file naming conventions prove to popular amongst the community.

Many thanks!
